src/com/amazon/ionschema/internal/ImportImpl.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.ionschema.internal
+
+import com.amazon.ionschema.Import
+import com.amazon.ionschema.Schema
+import com.amazon.ionschema.Type
+import java.lang.IllegalArgumentException
+
+/**
+ * Implementation of [Import] for all user-provided ISL.
+ */
+internal class ImportImpl(
+        override val id: String,
+        private val schema: Schema,
+        private val importEntireSchema: Boolean,
```

Comment:
Can we have `importEntireSchema = true` and `types` not empty and not `null`? Which .isl file as input would cause such a scenario?

Reply:
Hopefully not common usage, but [schema/import/import_types.isl](https://github.com/amzn/ion-schema-tests/blob/master/schema/import/import_types.isl) provides an example of this.  The entire positive_int.isl schema is imported, *and* the `positive_int` type is aliased to `posint` via a second import.  These two imports result in invoking this constructor with `importEntireSchema = true` and a non-empty `types` map.

Reply:
Hm, so if for the `importEntireSchema` case we implemeted it as if we added 'as' statements to alias to the same name then the implementation will only have to deal with aliases and should simplify the code. Unless there is a need to be aware that an import did not add aliases at all for some other reason

Reply:
I like the idea of fetching types from the Schema (as currently implemented in this PR); if at some point we allow types to be added to a Schema, a full import of the Schema would reflect those changes.

However, your comment has made me realize this approach presents a potential problem.  If my schema defines type A and imports all of schema X, if someone can dynamically add a new type A to schema X, which type A does my schema see?  We've avoided such problems up to this point, and the approach you're describing avoids inadvertently presenting this issue in the future.  Will take a closer look.

Reply:
Ok, I've simplified as you suggested, and `importEntireSchema` is no more.  This feels all-around safer and a bit simpler.